### PR TITLE
Road updates

### PR DESCRIPTION
**Pull request by Santonia27**
- [x] Add the method to the "road" type (centroid) in the HMT created exposure file

- [x] Change the units of all height/length to the units in the settings.toml --> this means the segment lengths as well, and remove unit from column heading

- [x] Change "roads" type to "road"

- [x] Remove the step function and the max damage of "1" - we are for now only using the calculated inundation depth on the roads

- [ ] (Panos) Revise the CHS model to have correct segment lengths (do with HMT after corrections)

### CODE REVIEW

File: hydromt_fiat/api/data_types.py [hunk 1]
```diff
@@ -105,18 +105,18 @@ class ExposureSetupDamages(BaseModel):
...
-    min_hazard_value: float
-    max_hazard_value: float
-    step_hazard_value: float
-    vertical_unit: Units
+#class RoadVulnerabilitySettings(BaseModel):
+    #threshold_value: float
+    #min_hazard_value: float
+    #max_hazard_value: float
+    #step_hazard_value: float
+    #vertical_unit: Units
```
Comment (panosatha) [inline hydromt_fiat/api/data_types.py:113]: why did you need to comment this out? I was expecting that you will only comment out only the part that uses the method to setup the roads (set_road_damage_threshold). 

But I don't see any problem for the actual classes and functions to still be there.

File: hydromt_fiat/api/exposure_vm.py [hunk 2]
```diff
@@ -286,7 +289,7 @@ def set_roads_settings(
         self.exposure_roads_model = ExposureRoadsSettings(
             roads_fn="OSM",
             road_types=road_types,
-            road_damage=1,
+            road_damage= math.nan,
```
Comment (panosatha) [inline hydromt_fiat/api/exposure_vm.py:292]: why are you using the math.nan ? I haven't seen this before. Why not "None" or "np.nan" if you know it is a float?

File: hydromt_fiat/api/hydromt_fiat_vm.py [hunk 3]
```diff
@@ -105,10 +105,10 @@ def build_config_yaml(self):
...
             config_yaml.setup_exposure_roads = self.exposure_vm.exposure_roads_model
 
-        if self.vulnerability_vm.vulnerability_roads_model:
-            config_yaml.setup_road_vulnerability = (
-                self.vulnerability_vm.vulnerability_roads_model
-            )
+        #if self.vulnerability_vm.vulnerability_roads_model:
+        #    config_yaml.setup_road_vulnerability = (
+        #        self.vulnerability_vm.vulnerability_roads_model
+        #    )
```
Comment (panosatha) [inline hydromt_fiat/api/hydromt_fiat_vm.py:111]: see my previous comment. Here the if statement should not pass if you don't have self.vulnerability_vm.vulnerability_roads_model right? So you could keep it the code.

File: hydromt_fiat/api/vulnerability_vm.py [hunk 4]
```diff
@@ -58,11 +58,11 @@ def save_linking_table(self) -> None:
...
-        )
\ No newline at end of file
+    #def set_road_damage_threshold(self, threshold: Union[int, float]):
+    #    self.vulnerability_roads_model = RoadVulnerabilitySettings(
+    #        threshold_value=threshold,
+    #        min_hazard_value=0,
+    #        max_hazard_value=16,
+    #        step_hazard_value=1.0,
+    #        vertical_unit=Units.feet.value,
+    #   )
```
Comment (panosatha) [inline hydromt_fiat/api/vulnerability_vm.py:68]: see previous comment

File: hydromt_fiat/workflows/exposure_vector.py [hunk 5]
```diff
@@ -267,24 +268,36 @@ def setup_roads(
...
+            roads["Extraction Method"] = "centroid"
         elif isinstance(road_damage, int):
-            roads["Segment Length [m]"] = get_road_lengths(roads)
+            roads["Segment Length"] = road_length
             roads["Max Potential Damage: Structure"] = road_damage
+            roads["Extraction Method"] = "centroid"
+        elif isinstance(road_damage, float):
+            roads["Segment Length"] = road_length
+            roads["Max Potential Damage: Structure"] = math.nan
+            roads["Extraction Method"] = "centroid"
```
Comment (panosatha) [inline hydromt_fiat/workflows/exposure_vector.py:300]: this part is not clear to me. For example why is this part `roads["Extraction Method"] = "centroid"` in the if statement since it is exactly the same every time?

Then why for road damage, there are option for it to be str, int or float? What does this mean? I think these functions need a bit of documentation as well.

**Review by panosatha — CHANGES_REQUESTED** [review]
I have a couple of comments on the code.
Moreover, I tested the model builder and the results looked fine. But I had two extra questions:

- I noticed that while there is not depth damage function for roads, there is still a value "road" in the "Damage Function: Structure" column.
- For some reason the roads with the quick build do not include all the road types anymore.

File: hydromt_fiat/workflows/exposure_vector.py (hunk 5 above)
Comment (Santonia27) [reply hydromt_fiat/workflows/exposure_vector.py:300]: > this part is not clear to me. For example why is this part `roads["Extraction Method"] = "centroid"` in the if statement since it is exactly the same every time?
> 
> Then why for road damage, there are option for it to be str, int or float? What does this mean? I think these functions need a bit of documentation as well.

In this case I did not change the code that was already existing, it was written like this before by someone else (with the options for int, float, str). I guess it is to cover all possibilities of user input road data? I put merged the int, float in one elif statement to only distinguish between str and (int,float) and moved the extraction method out of the if-statements

**Comment by Santonia27** [comment]
> I have a couple of comments on the code. Moreover, I tested the model builder and the results looked fine. But I had two extra questions:
> 
> * I noticed that while there is not depth damage function for roads, there is still a value "road" in the "Damage Function: Structure" column.
> * For some reason the roads with the quick build do not include all the road types anymore.

Do you mean it is not displaying the roads or there are actually roads missing in the csv?

**Review by panosatha — APPROVED** [review]
looks good
